cmd/snap/cmd_known.go

```diff
@@ -110,9 +118,20 @@ func (x *cmdKnown) Execute(args []string) error {
 
 	var assertions []asserts.Assertion
 	var err error
-	if x.Remote {
+	switch {
+	case x.Remote && !x.Direct:
+		// --remote will query snapd
+		assertions, err = x.client.Known(string(x.KnownOptions.AssertTypeName), headers, &client.KnownOptions{Remote: true})
+		// if snapd is unavailable automatically fallback
+		var connErr client.ConnectionError
```

Comment:
are authorisation errors also ConnectionErrors ? (I don't remember)

Reply:
authorization errors are client.Errors of kind ErrorKindLoginRequired. But we should not get one for "Known" because its userOk - I can update the code of course to deal with this.